ui/article-aem-renderer/src/main/java/org/cru/godtools/article/aem/db/AemImportDao.kt

```diff
@@ -12,9 +12,8 @@ import org.cru.godtools.article.aem.model.AemImport.AemImportArticle
 
 @Dao
 interface AemImportDao {
-    @get:WorkerThread
```

Comment:
This annotation was what was allowing it to run on the main thread? Is the issue because it's an IO operation and could take some time to finish, which is why doing it on the main thread would be a bad idea?

Reply:
The `@WorkerThread`, `@UiThread`, `@MainThread` are purely used by lint rules to try and detect improper thread usage of methods/objects/etc. They aren't foolproof though.

The main problem was that this was being called during creation of `AemArticleManager`, which could happen on the main thread in certain scenarios.

Reply:
converting the `Dao`/`Repository` methods to be suspend methods make them safe to call on any thread.